apis/datadoghq/v1alpha1/watermarkpodautoscaler_types.go

```diff
@@ -132,6 +132,37 @@ type WatermarkPodAutoscalerSpec struct {
 	ReadinessDelaySeconds int32 `json:"readinessDelaySeconds,omitempty"`
 }
 
+// TLSConfig specifies the Recommender http client TLS configuration
+// +k8s:openapi-gen=true
+type TLSConfig struct {
+	// CAFile is a path to a CA certificate
+	// +optional
+	CAFile string `json:"caFile,omitempty"`
+
+	// CertFile is a path to a client Cert
+	// +optional
+	CertFile string `json:"certFile,omitempty"`
+
+	// Keyfile is a path to the client certificate key (mandatory if CertFile is set)
+	// +optional
+	KeyFile string `json:"keyFile,omitempty"`
+
+	// ServerName is a settings to activate TLS SNI
+
+	ServerName string `json:"serverName,omitempty"`
+
+	// InsecureSkipVerify when true disable verifying server certificate
+	InsecureSkipVerify bool `json:"insecureSkipVerify,omitempty"`
+
+	// MinVersion, minimum TLS version, defaults to Go default which is TLS1.2
+	// +optional
+	MinVersion string `json:"minVersion,omitempty"`
```

Comment:
This is a new feature, we don't need to support anything else than 1.3. Similarly, if we dont have a use case for `ServerName` right now, I would drop it and potentially add later if needed.